Riot/Views/Authentication/ForgotPasswordInputsView.m

```diff
@@ -197,42 +197,73 @@ - (void)prepareParameters:(void (^)(NSDictionary *parameters))callback
+                                                   
+                                                   [strongSelf.nextStepButton addTarget:strongSelf
+                                                                                 action:@selector(didCheckEmail:)
+                                                                       forControlEvents:UIControlEventTouchUpInside];
+                                                   
+                                                   strongSelf.nextStepButton.hidden = NO;
+                                               }
+                                           } failure:^(NSError *error) {
+                                               NSLog(@"[ForgotPasswordInputsView] Failed to request email token");
+                                               
+                                               // Ignore connection cancellation error
+                                               if (([error.domain isEqualToString:NSURLErrorDomain] && error.code == NSURLErrorCancelled))
+                                               {
+                                                   return;
+                                               }
+                                               
+                                               NSString *errorMessage;
+                                               if (error.userInfo[@"error"])
+                                                   errorMessage = error.userInfo[@"error"];
+                                               else
+                                                   errorMessage = error.localizedDescription;
+                                               
+                                               __strong typeof(weakSelf) strongSelf = weakSelf;
+                                               if (strongSelf) {
+                                                   if (strongSelf->inputsAlert)
+                                                   {
+                                                       [strongSelf->inputsAlert dismiss:NO];
+                                                   }
+                                                   
+                                                   strongSelf->inputsAlert = [[MXKAlert alloc] initWithTitle:[NSBundle mxk_localizedStringForKey:@"error"] message:errorMessage style:MXKAlertStyleAlert];
+                                                   strongSelf->inputsAlert.cancelButtonIndex = [inputsAlert addActionWithTitle:[NSBundle mxk_localizedStringForKey:@"ok"] style:MXKAlertActionStyleDefault handler:^(MXKAlert *alert) {
+                                                       strongSelf->inputsAlert = nil;
+                                                       [strongSelf.delegate authInputsViewDidCancelOperation:strongSelf];
+                                                   }];
+                                                   
```

Comment:
@morozkin You're right, you have to call `authInputsViewDidCancelOperation` in case of error.

Please add the following check before calling the delegate: `if (strongSelf.delegate && [strongSelf.delegate respondsToSelector:@selector(authInputsViewDidCancelOperation:)])`. This method is optional.